docs/dev/object-deletion.md

```diff
@@ -0,0 +1,118 @@
+* `0000_90_cluster-svcat-apiserver-operator_00_prometheusrole.yaml` containing a Role in the `openshift-service-catalog-apiserver-operator` namespace.
+    This manifest would be dropped, because the Role would be removed as part of the namespace deletion.
+* `0000_90_cluster-svcat-apiserver-operator_01_prometheusrolebinding.yaml` containing a RoleBinding in the `openshift-service-catalog-apiserver-operator` namespace.
+    This manifest would be dropped, because the RoleBinding would be removed as part of the namespace deletion.
+* `0000_90_cluster-svcat-apiserver-operator_02-operator-servicemonitor.yaml` containing a ServiceMonitor in the `openshift-service-catalog-apiserver-operator` namespace.
+    This manifest would be dropped, because the ServiceMonitor would be removed as part of the namespace deletion.
+
+So the remaining manifests with deletion annotations would be the namespace and the cluster-scoped CRD, ServiceCatalogAPIServer, ClusterRoleBinding, and ClusterOperator.
+The ordering of the surviving manifests would not be particularly important, although keeping the namespace first would avoid removing the ClusterRoleBinding while the consuming Deployment was still running.
+If multiple deletions are required, it is up to the developer to name the manifests such that deletions occur in the correct order.
+
+Similar handling would be required for the svcat-controller-manager operator.
+
+If resources external to kubernetes must be removed the developer must provide the means to do so.
+This is expected to be done through modification of an operator to do the removals during it's finalization.
+If operator modification for object removal is necessary that operator would be deleted in a subsequent update.
+
+The deletion manifests described above would have been preserved through 4.5.z release and removed in 4.6.
+See the [Subsequent Releases Strategy](#subsequent-releases-strategy) section for details.
+
+## Removing functionality that users might notice
+
+Below is the flow for removing functionality that users might notice, like the web console.
+
+* The first step is deprecating the functionality.
+    During the deprecation release 4.y, the functionality should remain available, with the operator setting Upgradeable=False and linking release notes like [these][deprecated-marketplace-apis].
+* Cluster Administrators must follow the linked release notes to opt in to the removal before updating to the next minor release in 4.(y+1).
+    When the administrator opts in to the removal, the operator should stop setting Upgradeable=False.
+* The operand may be removed when Cluster Administrators opt-in to the removal, or it may be left running and be removed during the transition to the next minor release 4.(y+1).
+* The update to the next minor release 4.(y+1) may use the manifest delete annotation to remove the operator, and, if they have not already been removed, any remaining operand components.
```

Comment:
Can yo clarify the above sentence? It is not clear to me.

Reply:
done